docs/rules/no-named-import.md

```diff
@@ -0,0 +1,86 @@
+# Prevent named imports from react (react/no-named-import)
+
+Prevent the use of named imports from React, so it's visible in code that it's React and not custom code. This rule
+also creates consistency inside a codebase because some people use named imports and some don't.
+
+## Rule Details
+
+### Options
+
+This rule has two different properties that can be used (not at the same time): `allow` and `forbid`. Both are an array of strings. By default this rule allows nothing, so every named import will error.
+
+Example configurations:
+
+```javascript
+// forbid
+{
+  "rules": {
+    "react/no-named-import": ["error", {
+      "forbid": ["Component", "useEffect"],
+      // Optional
+      forceImport: true,
```

Comment:
i'm a bit skeptical about this one; when this is `true`, then imports are forced, but when it's set to `false`, are properties forced?

Reply:
`
// { allow: ['useState'], forceImport: true }
import React from 'react';
// will error, because it will force to import useState
const [loading, setLoading] = React.useState(false);
`

`
// { allow: ['useState'], forceImport: false }
import React from 'react';
// will not error, because it will allow the user to use React.useState
const [loading, setLoading] = React.useState(false);
`

`
// { allow: ['useState'], forceImport: false }
// will error on the Component import, because that's not allowed
import React, { Component } from 'react';
const [loading, setLoading] = React.useState(false);
`

Reply:
ok, so then it seems like the entire rule has two "modes" - import, or property. If that's the case, then that shouldn't be in the options bag, it should be the first enum argument in a 2-argument schema (the object comes second). Does that make sense?

Reply:
I get what you wrote, only that's not the current implementation, because the `forceImport` property doesn't force a mode (property/import), let me clarify using these examples:

```
// allow ['useState']
// forceImport: false (= default value)

import React from 'react';

// passes, because forceImport === false
const [loading, setLoading] = React.useState(false);

// passes, because allow['useState']
import React, { useState } from 'react';

// errors, because Component is not allowed to import
import React, { Component } from 'react';

// passes, because Component is not included in allow
class MyComponent extends React.Component {}
``` 

Another example:
```
// allow['useState']
// forceImport: true

import React from 'react';
// error, because forceImport: true, useState should be imported
const [loading, setLoading] = React.useState(false);

// Like this:
import React, { useState } from 'react';
const [loading, setLoading] = useState(false);
```

with the forbid option:
```
// forbid: ['Component']
// forceImport: true

import React from 'react';
// error, forceImport: true, so all except Component should be imported
const [loading, setLoading] = React.useState(false);

// forbid: ['Component']
// forceImport: false
import React from 'react';

// passes, because forceImport: false. User is allowed to access property and use it
const [loading, setLoading] = React.useState(false);

// errors, because Component is forbidden to import
import React, { Component } from 'react';
```

Reply:
It seems like you envision `forceImport false` to mean "ignore" where `forceImport true` means "force imports" - where i'm looking for "force properties". iow, it seems like this shouldn't be a boolean `forceImport` but should instead be an enum `style` or something with 3 values?